<launch>

  <arg name="serial" default="$(arg serial)"/>
  <arg name="camera_name" default="bluefox3_$(arg serial)"/>
  <arg name="frame_id" default="$(arg camera_name)"/>
  <arg name="calib_url" default="file://$(find bluefox3)/config/calib/bluefox3_FF005889.yaml"/>

  <arg name="debug" default="false" />

  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <!-- Node Settings -->
  <arg name="plugin" default="false"/>
  <arg name="manager" default="bluefox3_manager"/>
  <arg name="output" default="screen"/>
  <arg name="proc" default="false"/>

  <!-- Standalone -->
  <arg unless="$(arg plugin)" name="option" value="standalone bluefox3/Bluefox3"/>

  <!-- Manager -->
  <arg if="$(arg plugin)" name="option" value="load bluefox3/Bluefox3 $(arg manager)"/>

  
  <!--<remap from="/bluefox3_F0800081/image_raw" to="/camera/image_raw" />-->

  <!-- Nodelet -->
  <node pkg="nodelet" type="nodelet" name="$(arg camera_name)" args="$(arg option)" output="$(arg output)" respawn="false" launch-prefix="$(arg launch_prefix)">

    <rosparam file="$(find bluefox3)/config/bluefox3.yaml" />

    <param name="exposure_time" type="string" value="1000"/>
    <param name="exposure_auto" type="string" value="Off"/>
    <param name="exposure_auto_upper_limit" type="double" value="25000.0"/>
    <param name="exposure_auto_lower_limit" type="double" value="10000.0"/>
    <param name="camera_serial" type="string" value="$(arg serial)"/>
    <param name="camera_name" type="string" value="$(arg camera_name)"/>
    <param name="frame_id" type="string" value="$(arg frame_id)"/>
    <param name="calib_url" type="string" value="$(arg calib_url)"/>

    <!-- Disable publishing of specific image transport topics -->
    <rosparam param="image_raw/disable_pub_plugins">
    - 'image_transport/compressedDepth'
    - 'image_transport/theora'
    - 'image_transport/h264'
    </rosparam>

    <!-- Publishers -->   

  </node>


  <node pkg="decimate_image" type="decimate_image_node" name="decimate_image_name"  output="$(arg output)" respawn="false">

  </node>

</launch>

